<a href="https://colab.research.google.com/github/antvilla06/numerico/blob/master/Curso%20M%C3%A9todos%20Num%C3%A9ricos%20Unidad_I%20.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Introducción


In [ ]:
5+6


11

In [ ]:
7+8

15

Si suman $1.1+2.2=3.3$ 

In [ ]:
1.1+2.2

3.3000000000000003

El 1.1 en representación binaria tiene infinitos 0 y 1 en su expansio.

Tipos de datos en Python.

Números Flotantes: Son representaciones de los números reales en un lengaje de programación. Los números pueden dar patologias, tal es el caso de sumar 

  $1.1+2.2=3.3000000000000003$

  Ejemplos de numeros flotantes:

          pi = 3.141519.....(infinitas cifras decimales no periodicas)
          1/3= 0.33333333333.....
          e =2.718.........(infinitas cifras decimales no periodocas)


Notación exponencial: Cuando un número es muy grande se repserenta en forma exponencial. 

  $4000 = 4 *1000=4*10^3= 4e3 $ (notación de python)

  $5e2=5*10^2=500$

  $60000=6*10000=6*10^4=6e4$ 

  $5.678e6=5.678*10^6=5.678*1000000=5678000$ 

In [ ]:
x = 5e2
print(x)

z = 6e4
print(z)

w = 5.678e6
print(w)

500.0
60000.0
5678000.0


Tipo de datos 2: Boloanos. Son proposiciones lógicas que dan condiciones para realizar teareas indicadas en la computadora.

          el condiconal "if" permite dar una condicoón para realizar una tarea. 

Números enteros y decimales: Son representaciones enteras de cantidades de números flotantes (reales) y los decimales son representaciones finitas de números fotantes

.

Tipo de datos 3 Arrelgos o array: Son representaciones en forma de "array" o vector.

In [ ]:
import numpy as np
#Representación de arreglos (arrays) 

A=[1,2,3,4]
print(A)

[1, 2, 3, 4]


Nota importante: Python (cualquier lengiaje de programación) no tiene la capaidad de indentificar variables  por si solas, siempre le tenemos que dar un valor.

Cadenas de caracteres (Strings): Son secuencias de caracteres. 

Python no puede reconoce letras por si solas, para darle sentido se encierran en "".

Se usan principalmente en la recopilación de datos y analisis de información.

In [ ]:
x="casa"
print(x)

casa


In [ ]:
#Diferencia hay entre 1 y 1.001


1.001**100

1.105115697720756



```
# Esto tiene formato de código
```

# Operadores de Asignación

Saber diferenciar entre operadores de asignación como por ejemplo el signo $=$ y $==$

In [ ]:
x = 5
print(x)

5


El signo $=$ asigna valores númericos a los objetos.

$x = 7$ le da valor $7$ a la variable $x$.

In [ ]:
2*x == 10

True

In [ ]:
5 == 5

True

In [ ]:
7 ==6

False

# Operadores de relación.

El signo $==$ sirve para comparar dos valores númericos.

Operadores de relación.

$7 != 5$ indics que los valores no son iguales.

$7 < 8$ indica que $7$ es menor que $8$.

$ 5 > 4$ indica que $5$ es mayor que $4$.

$7 <= 7$ indica que un número es menor o igual que el otro.



In [ ]:
7 <9

True

In [ ]:
6 != 6

False

In [ ]:
7 <= 7

True

In [ ]:
6 <= 7

True

In [ ]:
16 >= 20

False

# Operadores de asignación.

In [ ]:
r = 10
r += 5
print(r)

15


El signo $r+=5$ suma el varlor 5 a la variable $r$.

Otra forma equivalente es escribir $r=r+5$


In [ ]:
z = 14

z = z +4

print(z)

18


El operador $r -= 8$  le resta $8$ a $r$.

El equivalente es $r=r-8$*texto en cursiva*

In [ ]:
m = 34

m -= 8
print(m)

26


$r *= 7$ multiplca $7$ por $r$.

$r \= 9$ dividir el valor de $r$ entre $p$.

$r **=5$ elevar el valor de $r$ a la $5$.

In [ ]:
f = 10
f *= 7
#Significa que f =7*f
print(f) 

70


In [ ]:
s = 3
s **= 4
#Significa que s = s**4
print(s)

81


#Boleanos

Los boleanos son las cadenas True y False

La instruccción int() le asigna valor númerico a True y False.



In [ ]:
int(True)

1

In [ ]:
int(False)

0

In [ ]:
False

False

In [ ]:
False == False

True

In [ ]:
True != True

False

In [ ]:
() == False

False

In [ ]:
() != False

True

In [ ]:
{} == 0

False

In [ ]:
() == 1

False

In [ ]:
() == 0

False

In [ ]:
{} == False

False

In [ ]:
0 == False

True

In [ ]:
1 == False

False

La tarea bool()

In [ ]:
bool(6)

True

In [ ]:
bool({})

False

In [ ]:
bool(False)

False

In [ ]:
bool(True)

True

In [ ]:
bool(1)

True

In [ ]:
bool(0)

False

# Sección nueva

In [ ]:
A= [7,6,8,9]
print(A)



[7, 6, 8, 9]


In [ ]:
A = []
A.append(6)
print(A)
A.append(7)
print(A)
A.append(9)
print(A)

[6]
[6, 7]
[6, 7, 9]
